modem: hl7800: fix power off

rerickson1 opened the pull request:
Ensure DNS resolver is stopped if modem is turned off.
All IO connected to the HL7800 need to be de-asserted
in order for the HL7800 to power off.

fixes https://github.com/zephyrproject-rtos/zephyr/issues/54288

ycsin commented on drivers/modem/hl7800.c:
```diff
@@ -5468,29 +5484,53 @@ int32_t mdm_hl7800_reset(void)
-	wakeup_hl7800();
-	hl7800_stop_rssi_work();
+	LOG_DBG("Shutdown DNS resolver");
+	dnsCtx = dns_resolve_get_default();
```
not sure if this camelCase is intended?

rerickson1 replied:
Fixed, this was a historical copy paste thing.

rerickson1 committed:
modem: hl7800: fix power off

Ensure DNS resolver is stopped if modem is turned off.
All IO connected to the HL7800 need to be de-asserted
in order for the HL7800 to power off.

Signed-off-by: Ryan Erickson <ryan.erickson@lairdconnect.com>
